eZ/Publish/Core/MVC/Symfony/Security/Authentication/RememberMeRepositoryAuthenticationProvider.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+/**
+ * File containing the RememberMeRepositoryAuthenticationProvider class.
+ *
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace eZ\Publish\Core\MVC\Symfony\Security\Authentication;
+
+use eZ\Publish\API\Repository\Repository;
+use Symfony\Component\Security\Core\Authentication\Provider\RememberMeAuthenticationProvider;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+
+class RememberMeRepositoryAuthenticationProvider extends RememberMeAuthenticationProvider
+{
+    /**
+     * @var \eZ\Publish\API\Repository\Repository
+     */
+    private $repository;
+
+    public function setRepository(Repository $repository)
+    {
+        $this->repository = $repository;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function authenticate(TokenInterface $token)
+    {
+        $authenticatedToken = parent::authenticate($token);
+        if (empty($authenticatedToken)) {
+            return;
```

Comment:
@alongosz I inherit from Symfony `RememberMeAuthenticationProvider` and this does return void. Yes, that is inconsistent with doc, but that's Symfony's inconsistency. I'm being consistent with their inconsistency :)
https://github.com/symfony/symfony/blob/2.8/src/Symfony/Component/Security/Core/Authentication/Provider/RememberMeAuthenticationProvider.php#L43

The parent calls `supports()` so I found it redundant to do the same. Only one would need to actually execute, but still. Since I use their `authenticate()` I would rather rely on their logic, whatever it is, rather than repeat or try to improve it on my end.

I could throw `AuthenticationException` instead of returning void as Symfony does. Would that be OK?

Reply:
I could also send Symfony a PR, might be fun :) Their implementation does seem to be in direct contradiction to the interface doc, indeed.

Reply:
Other implementations of `Symfony\Component\Security\Core\Authentication\Provider\AuthenticationProviderInterface` are also consistent in this inconsistency :wink: So for me, this is a bug is in docblock, rather than in implementation itself.

Reply:
Uh :)
If `null` / `void` is really an unexpected state (meaning: not a result of some early binding of components), then it is fine to throw an exception.

I point out such things because if we'll use strict PHP7 types in the future (somewhere in the next millennium ;) ), technical debt would be impossible to handle.

Reply:
Right. Well, whether null is expected or not depends on the outcome of the symfony doc PR
https://github.com/symfony/symfony/pull/24585

I'm actually uncertain if throwing an exception would be correct here. It could be that Symfony depends on the null to proceed to trying other auth providers, for all I know. If so, omitting it could lead to weird bugs, depending on provider order. Just speculating.

Reply:
So far it seems Symfony reviewers would rather make auth providers throw, than changing the doc. And that change is perfectly safe, as the auth provider manager actually checks `supports()` before calling `authenticate()` so the throw Should Never Happen(TM) anyway.

Reply:
@alongosz Throwing now, since that seems to be what the Symfony team wants.

Reply:
@alongosz Still doubtful? :)

Reply:
> @alongosz Still doubtful? :)

Nope, this is perfect. Got lost in flood of review requests ;)